src/openssl/lib_crypto.cr

```diff
@@ -1,9 +1,23 @@
 {% begin %}
   lib LibCrypto
-    {% from_libressl = (`hash pkg-config 2> /dev/null || printf %s false` != "false") &&
-                       (`test -f $(pkg-config --silence-errors --variable=includedir libcrypto)/openssl/opensslv.h || printf %s false` != "false") &&
-                       (`printf "#include <openssl/opensslv.h>\nLIBRESSL_VERSION_NUMBER" | ${CC:-cc} $(pkg-config --cflags --silence-errors libcrypto || true) -E -`.chomp.split('\n').last != "LIBRESSL_VERSION_NUMBER") %}
-    {% ssl_version = `hash pkg-config 2> /dev/null && pkg-config --silence-errors --modversion libcrypto || printf %s 0.0.0`.split.last.gsub(/[^0-9.]/, "") %}
+    {% if flag?(:win32) %}
+      {% from_libressl = false %}
+      {% ssl_version = nil %}
+      {% for dir in Crystal::LIBRARY_PATH.split(';') %}
+        {% unless ssl_version %}
+          {% config_path = "#{dir.id}\\openssl_VERSION" %}
+          {% if config_version = read_file?(config_path) %}
+            {% ssl_version = config_version.chomp %}
+          {% end %}
+        {% end %}
+      {% end %}
+      {% ssl_version ||= "0.0.0" %}
+    {% else %}
```

Comment:
I'd suggest to keep this complex macro logic out of this PR for now. As a minimal solution, we can just use an environment variable for now. That's sufficient for CI.

Storing the version information in a file is a distribution feature. How we implement such configuration should be discussed separately.

Reply:
#11492